src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`2.IVsImmediateStatementCompletion2.cs

```diff
@@ -26,6 +25,7 @@ int IVsImmediateStatementCompletion2.EnableStatementCompletion(int enable, int s
             if (filters.TryGetValue(textView, out var filter))
             {
                 filter.Enabled = enable != 0;
+                filter.WpfTextView.Properties[AsyncCompletionService.IsCompletionEnabled] = filter.Enabled;
```

Comment:
please maintain the status of completion feature here in this class, and call `featureService.Disable` right here